### PR TITLE
Move command line parameter code

### PR DESCRIPTION
**Pull request by IgorDeepakM**
Command line paramater code moved from os.nim to cmdparam.nim

### CODE REVIEW

**Comment by metagn** [comment]
I like the idea but there is no deliberation on this that I know of.

* should the module name be `cmdparam`?
* why is `parseCmdLine` here? if it's here, shouldn't stuff like `quoteShell` also be here? I think these would warrant another module (maybe with a typesafe API)

Would be nice to hear your reasoning.

Also new modules tend to go in lib/std, not lib/pure.

**Comment by IgorDeepakM** [comment]
> I like the idea but there is no deliberation on this that I know of.
> 
>     * should the module name be `cmdparam`?
> 
>     * why is `parseCmdLine` here? if it's here, shouldn't stuff like `quoteShell` also be here? I think these would warrant another module (maybe with a typesafe API)
> 
> 
> Would be nice to hear your reasoning.
> 
> Also new modules tend to go in lib/std, not lib/pure.

When it comes to the module name cmdparam, this was my idea of the name as the module facilitates obtaining och parsing the command line. This can be changed if you have other suggestions.

I put cmdparam.nim in lib/pure because os.nim also is located there. I'm not too familiar with the source code structure and what goes where. cmdparam.nim can be moved to lib/std if that's a better location.

When it comes to the `quoteShell` function, I didn't include these because these belong more to execution of external processes. `parseCmdLine` is used more for what commands you receive from the command line. 

Just to clarify the background and reason for this change. In order to use command line parameters you need to include the os module and this brings in the whole lot with many functionalities that might not even exist for the particular embedded system. For embedded systems, everything might not be implemented. Instead you can single out the command line functionality from a separate module by "import std/cmdparam". The goal is that you should be able to cherry pick functionality more rather than a big os module monolith.

**Comment by Araq** [comment]
The refactoring is great but the module's name should be `std / cmdline`.

**Comment by IgorDeepakM** [comment]
One check was cancelled after 60 minutes. There is not much information why this test failed other than the text.

```
Test selected Nimble packages
  testament --batch:0_3 pcat nimble-packages
  Downloading Official package list
      Success Package list downloaded.
  Error: The operation was canceled.
```

Any idea what this is about?

**Comment by ringabout** [comment]
Never mind. The test sometimes fails for unknown reasons. I have restarted it.

File: lib/std/cmdline.nim [hunk 1]
```diff
@@ -0,0 +1,313 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2015 Andreas Rumpf
```
Comment (ringabout) [inline lib/std/cmdline.nim]: The copyright year can be updated since it is a new module.

**Comment by metagn** [comment]
Do we add an extra `std/shell` for stuff like `quoteShell` or should they go here?

Also you don't have to bother manually squashing & force pushing every time, it's done when merging anyway

**Comment by IgorDeepakM** [comment]
> Do we add an extra `std/shell` for stuff like `quoteShell` or should they go here?
> 
> Also you don't have to bother manually squashing & force pushing every time, it's done when merging anyway

`quoteShell` can go to a separate module like std/shell, which is possible. However, does that belong to this PR?

I tried to keep it clean on my local branch, that's why I used amend commits with force push. I was not sure how smart Github is when it comes to magically remove merge commits  and subsequence change commits.

**Comment by metagn** [comment]
Yes it's fine if another new module is not part of this PR.

**Comment by rockcavera** [comment]
I believe there is a changelog entry missing.

**Comment by metagn** [comment]
I think the OS splits will have a change log entry at the end to avoid conflicts

**Comment by ringabout** [comment]
> I think the OS splits will have a change log entry at the end to avoid conflicts

@IgorDeepakM Nice work! It would be appreciated if you can add it to changelogs following 

> Added new modules which were part of std/os

**Comment by ringabout** [comment]
=> https://github.com/nim-lang/Nim/pull/21039

**Pull request merged**